File: 5f9aba12-2f21-4ed8-8aa6-9283a9ab9fe0/src/main/java/com/sportsevent/entity/Notification.java
package com.sportsevent.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Document(collection = "notifications")
public class Notification {

    @Id
    private String id;

    private NotificationType type;

    private String title;

    private String content;

    private String relatedEntityId;

    private String relatedEntityType;

    private List<Recipient> recipients;

    private NotificationChannel channel;

    private NotificationStatus status;

    private Integer retryCount;

    private LocalDateTime scheduledAt;

    private LocalDateTime sentAt;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    public enum NotificationType {
        SCHEDULE_CHANGE, VENUE_CHANGE, SCORE_PUBLISHED, REGISTRATION_RESULT,
        MATCH_REMINDER, REFEREE_ASSIGNMENT, LEAGUE_ANNOUNCEMENT, APPEAL_RESULT
    }

    public enum NotificationChannel {
        IN_APP, WECHAT_TEMPLATE, SMS, EMAIL
    }

    public enum NotificationStatus {
        PENDING, SENT, FAILED, DELIVERED, READ
    }

    @Data
    public static class Recipient {
        private String recipientId;
        private RecipientType recipientType;
        private String name;
        private String contact;
        private NotificationStatus individualStatus;
        private LocalDateTime deliveredAt;

        public enum RecipientType {
            ATHLETE, TEAM, REFEREE, COORDINATOR, VENUE_MANAGER
        }
    }
}

File: 9096d8e2-d12b-468a-91ae-53c97e7bf587/backend/src/main/kotlin/com/ems/dispatch/entity/Notification.kt
package com.ems.dispatch.entity

import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Entity
@Table(name = "notifications")
@EntityListeners(AuditingEntityListener::class)
data class Notification(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "notification_no", nullable = false, unique = true, length = 50)
    var notificationNo: String,

    @Column(nullable = false, length = 30)
    var type: String,

    @Column(nullable = false, length = 20)
    var priority: String = "NORMAL",

    @Column(nullable = false, length = 200)
    var title: String,

    @Column(columnDefinition = "TEXT")
    var content: String? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dispatch_event_id")
    var dispatchEvent: DispatchEvent? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipient_user_id")
    var recipientUser: User? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipient_hospital_id")
    var recipientHospital: Hospital? = null,

    @Column(nullable = false, length = 30)
    var channel: String = "WEBSOCKET",

    @Column(nullable = false, length = 20)
    var status: String = "PENDING",

    @Column(name = "sent_at")
    var sentAt: LocalDateTime? = null,

    @Column(name = "read_at")
    var readAt: LocalDateTime? = null,

    @Column(name = "ack_received")
    var ackReceived: Boolean = false,

    @Column(name = "ack_at")
    var ackAt: LocalDateTime? = null,

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    var createdAt: LocalDateTime = LocalDateTime.now()
) {
    enum class NotificationType {
        NEW_DISPATCH, DISPATCH_UPDATE, VEHICLE_ALERT, MEDICAL_RECORD_SUBMITTED,
        QC_REVIEW_REQUIRED, HOSPITAL_NOTIFICATION, MAINTENANCE_ALERT,
        SUPPLY_LOW_ALERT, SYSTEM_ALERT, USER_NOTIFICATION
    }

    enum class Priority {
        LOW, NORMAL, HIGH, URGENT, EMERGENCY
    }

    enum class Channel {
        WEBSOCKET, SMS, EMAIL, APP_PUSH, IN_APP
    }

    enum class Status {
        PENDING, SENDING, SENT, DELIVERED, READ, FAILED, ACKNOWLEDGED
    }
}
